core/arch/arm/mm/core_mmu.c

```diff
@@ -506,7 +526,15 @@ static void init_mem_map(struct tee_mmap_region *memory_map, size_t num_elems)
 		add_phys_mem(memory_map, num_elems, &m, &last);
 	}
 
-	verify_sdp_mem_areas(memory_map, num_elems);
+#ifdef CFG_SECURE_DATA_PATH
+	verify_special_mem_areas(memory_map, num_elems,
+				 &__start_phys_sdp_mem_section,
+				 &__end_phys_sdp_mem_section, "SDP");
+#endif
+
+	verify_special_mem_areas(memory_map, num_elems,
+				 &__start_phys_nsec_ddr_section,
+				 &__end_phys_nsec_ddr_section, "NSEC DDR");
```

Comment:
You propose to force disable nsec ddr regions when SDP is enabled?
I have created function that cross-checks SDP and NSEC DDR regions. I thing it is better approach.